### PR TITLE
feature/6429-Dylan-InvalidRoutingNumberValidation

### PR DESCRIPTION
**Pull request by Sparowhawk**
## Screenshots/Video
Before had no validation it would just save. Here is post validation

too few numbers
Screenshot 2023-08-16 at 4 06 26 PM

non valid routing number
Screenshot 2023-08-16 at 4 04 15 PM

valid routing number
Screenshot 2023-08-16 at 4 03 56 PM

## Testing
yarn text

- [x] Tested on iOS 
- [x] Tested on Android 

## Reviewer Validations
 update front-end routing number form validation
 display the pre-existing error when invalid routing number is supplied

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [x] PR is connected to issue(s)
- [x] Tests are included to cover this change (when possible)
- [x] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [x] No secrets or API keys are checked in
- [x] All imports are absolute (no relative imports)
- [x] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/src/screens/PaymentsScreen/DirectDepositScreen/EditDirectDepositScreen/EditDirectDepositScreen.tsx [hunk 1]
```diff
@@ -137,6 +137,21 @@ const EditDirectDepositScreen: FC<EditDirectDepositProps> = ({ navigation, route
     return !/^\d+$/.test(input)
   }
 
+  const validRoutingNumber = (input: string): boolean => {
```
[the remark this answers is not in the document]
Comment (Sparowhawk) [reply VAMobile/src/screens/PaymentsScreen/DirectDepositScreen/EditDirectDepositScreen/EditDirectDepositScreen.tsx]: updated

File: VAMobile/src/screens/PaymentsScreen/DirectDepositScreen/EditDirectDepositScreen/EditDirectDepositScreen.tsx [hunk 2]
```diff
@@ -137,6 +137,21 @@ const EditDirectDepositScreen: FC<EditDirectDepositProps> = ({ navigation, route
     return !/^\d+$/.test(input)
   }
 
+  const invalidRoutingNumber = (input: string): boolean => {
```
Comment (theodur) [inline VAMobile/src/screens/PaymentsScreen/DirectDepositScreen/EditDirectDepositScreen/EditDirectDepositScreen.tsx:140]: This function says `067014822` is an invalid routing number, even though it's actually valid. I noticed the logic in one of the [links](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/6429#issuecomment-1664242741) that Tim shared resembles the [logic](https://github.com/department-of-veterans-affairs/va-mobile-app/issues/6429#issuecomment-1664406416) that EVSS is using. We'll probably want to use one of those approaches instead

**Review by theodur — CHANGES_REQUESTED** [review]

File: VAMobile/src/screens/PaymentsScreen/DirectDepositScreen/EditDirectDepositScreen/EditDirectDepositScreen.tsx (hunk 2 above)
Comment (Sparowhawk) [reply VAMobile/src/screens/PaymentsScreen/DirectDepositScreen/EditDirectDepositScreen/EditDirectDepositScreen.tsx:140]: @theodur that is the logic I used is the one that Tim and EVSS is using.

The logic is that numbers 1, 4, 7 are multiplied by 3, numbers 2, 5, 8 are multiplied by 7, and finally 3, 6, 9 are multiplied by 1. There was a typo that I was multiplying 1, 4, 7 and 2, 5, 8 by 3. I have fixed that and it now works.

**Review by dumathane — DISMISSED** [review]

**Review by theodur — APPROVED** [review]
Updates to the validation seem to be working fine now

**Review by theodur — APPROVED** [review]

**Review by dumathane — DISMISSED** [review]

**Review by theodur — APPROVED** [review]

**Pull request merged**